From 5e864385d4a5e8816eaaa43c35935c5dbc72125e Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Fri, 15 May 2020 18:18:44 +0800
Subject: [PATCH] gcc-10: disable 'array-bounds' warning for now

Build failure log:
arch/x86/page.c: In function 'init_ept_mem_ops':
arch/x86/page.c:240:48: error: array subscript is outside array bounds of 'struct page[0][1]' [-Werror=array-bounds]
240 | ept_pages_info[vm_id].ept.nworld_pml4_base = pre_uos_nworld_pml4_pages[vm_id];

Suppress the warning until fix is not available in upstream.

Upstream-Status: Pending

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 hypervisor/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hypervisor/Makefile b/hypervisor/Makefile
index 45a45c9e..4b9a22ae 100644
--- a/hypervisor/Makefile
+++ b/hypervisor/Makefile
@@ -79,6 +79,7 @@ CFLAGS += -m64 -mno-mmx -mno-sse -mno-sse2 -mno-80387 -mno-fp-ret-in-387
 CFLAGS += -mno-red-zone -mpopcnt
 CFLAGS += -nostdinc -nostdlib -fno-common
 CFLAGS += -Werror
+CFLAGS += -Wno-array-bounds
 CFLAGS += -O2
 ifeq (y, $(CONFIG_RELOC))
 CFLAGS += -fpie
-- 
2.17.1

